### PR TITLE
Use switch expressions in EnableAllocationDecider and NodeShutdownAllocationDecider

### PR DESCRIPTION
**Pull request by arteam**

### CODE REVIEW

**Comment by arteam** [comment]
@elasticmachine 
run elasticsearch-ci/part-1
run elasticsearch-ci/bwc

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/EnableAllocationDecider.java [hunk 1]
```diff
@@ -193,36 +178,16 @@ public Decision canRebalance(ShardRouting shardRouting, RoutingAllocation alloca
...
-            default:
-                throw new IllegalStateException("Unknown rebalance option");
-        }
+        return switch (enable) {
+            case ALL -> allocation.decision(Decision.YES, NAME, "all rebalancing is allowed");
+            case NONE -> allocation.decision(Decision.NO, NAME, "no rebalancing is allowed due to %s", setting(enable, usedIndexSetting));
+            case PRIMARIES -> shardRouting.primary()
+                ? allocation.decision(Decision.YES, NAME, "primary rebalancing is allowed")
+                : allocation.decision(Decision.NO, NAME, "replica rebalancing is forbidden due to %s", setting(enable, usedIndexSetting));
+            case REPLICAS -> shardRouting.primary() == false
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/EnableAllocationDecider.java:187]: Maybe invert this condition?

**Review by DaveCTurner — APPROVED** [review]
LGTM, one suggestion

**Comment by arteam** [comment]
Thanks David!

**Comment by DaveCTurner** [comment]
Oh I don't think this should have been backported to 8.1, it's not a bugfix. It was only labelled as 8.2 when I LGTMd it. Would you revert that backport please?

**Comment by arteam** [comment]
Sure, no problem :+1:

**Pull request merged**